tests/Collection/CollectionFunctionalTest.php

```diff
@@ -180,6 +182,41 @@ public function testWithOptionsPassesOptions()
         $this->assertSame(WriteConcern::MAJORITY, $debug['writeConcern']->getW());
     }
 
+    public function testMapReduce()
+    {
+        $this->createFixtures(3);
+
+        $map = new Javascript('function() { emit(this._id, this.x); }');
+        $reduce = new Javascript('function(key, values) { return Array.sum(values); }');
```

Comment:
I might suggest emitting on `1` instead of `this._id`, so that we collect all documents into a single bucket. Then, the reduce function can compute the sum of all `x` values. We can later assert that the result of map reduce is a single document with `[ '_id' => 1, 'x' => 66 ]`. Something like:

```
$expected = [
    [ '_id' => 1, 'x' => 66 ],
];

$this->assertSameDocuments($expected, $mapReduceResult);
```